skyvern/forge/agent.py

```diff
@@ -1773,6 +1774,24 @@ async def update_step(
             step_id=step.step_id,
             diff=update_comparison,
         )
+
+        try:
+            log = skyvern_context.current().log
+            current_step_log = [entry for entry in log if entry.get("step_id", "") == step.step_id]
```

Comment:
this PR is a good start.

There are a couple of problems i'm seeing right now (not because of the code here) that we need keep iterating:
- sometimes we don't have step_id in a log. [example](https://github.com/Skyvern-AI/skyvern/blob/759187354633f37085badcd224e76cd795fb2af2/skyvern/forge/agent.py#L273-L276). TODO: We need to add step_id to all the logs
- for each task in a workflow run, we don't set the task_id in the SkyvernContext before the task starts. this means many logs won't have task_id - if we want to store logs according to the "task_id" field in the json. TODO: set task_id in SkyvernContext before a task starts and reset it after a task ends (completed, failed, terminated, canceled)

Reply:
> TODO: We need to add step_id to all the logs

@wintonzheng are we certain that each log must have a step_id?

From what I see in the logs when I ran tasks there are several logs in the beginning that don't belong to any step:
```
2024-12-09T11:09:09.220558Z [info     ] Created new task               organization_id=o_333794107196816612 proxy_location=RESIDENTIAL task_id=tsk_335262355392558810 url=https://www.nytimes.com/books/best-sellers
2024-12-09T11:09:09.220672Z [info     ] Executing task using background task executor task_id=tsk_335262355392558810
2024-12-09T11:09:09.266772Z [info     ] Creating browser state for task task_id=tsk_335262355392558810
2024-12-09T11:09:12.092991Z [info     ] browser console log is saved   log_path=./log/2024-12-09/95733df0-f2d8-428d-89ce-f71221f0cbe1.log
2024-12-09T11:09:12.326437Z [info     ] Trying to navigate to https://www.nytimes.com/books/best-sellers and waiting for 5 seconds. retry_time=0 url=https://www.nytimes.com/books/best-sellers
2024-12-09T11:09:14.148604Z [info     ] Page loading time              loading_time=1.8220469951629639 url=https://www.nytimes.com/books/best-sellers
2024-12-09T11:09:19.148456Z [info     ] Successfully went to https://www.nytimes.com/books/best-sellers retry_time=0 url=https://www.nytimes.com/books/best-sellers
2024-12-09T11:09:19.162015Z [info     ] Starting agent step            step_id=stp_335262355392558812 step_order=0 step_retry=0 task_id=tsk_335262355392558810
```

Here first 7 logs would be without a step because the first step is not created yet.

Reply:
@wintonzheng do I undestand correctly that we want to create task and step ids in advance?

This way the logs that are currently not associated with a step or task because step/task not created yet  would be associated with the first task and first step?

Reply:
> @wintonzheng are we certain that each log must have a step_id?

no, step_id needs to be manually defined in `LOG.info("xxx", step_id="stp_xxxx")` nowadays.  those logs you shared here don't belong to any step so it's okay many logs don't have `step_id`.

> @wintonzheng do I undestand correctly that we want to create task and step ids in advance? This way the logs that are currently not associated with a step or task because step/task not created yet would be associated with the first task and first step?

yep, exactly what i mean. we need to backfill step_id and task_id in some logs.